### PR TITLE
Implemented MessagesResource

### PR DESCRIPTION
**Pull request by ricamgar**
Implemented the following methods of MessagesResource:
- getConversationByUserId()
- createConversation()
- sendMessageToConversation()

Added gradle task to install library locally (for testing purposes)

To review:
- [x] @serj-lotutovici / @dhartwich1991

### CODE REVIEW

**Comment by dhartwich1991** [comment]
Looks good to me :) Can you verify if we don't need to have the Copyright header in every file?

File: api-client/src/main/java/com/xing/api/data/messages/Conversation.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
+package com.xing.api.data.messages;
```
Comment (serj-lotutovici) [inline api-client/src/main/java/com/xing/api/data/messages/Conversation.java:17]: Need licence header.

File: api-client/src/main/java/com/xing/api/data/messages/Conversation.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        }
+
+        return (id.equals(c.id));
+    }
+
+    @Override
+    public int hashCode() {
+        return id != null ? id.hashCode() : 0;
+    }
+
```
Comment (serj-lotutovici) [inline api-client/src/main/java/com/xing/api/data/messages/Conversation.java:65]: Please provided an override for `toString()`

File: api-client/src/main/java/com/xing/api/data/messages/ConversationMessage.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
+package com.xing.api.data.messages;
```
Comment (serj-lotutovici) [inline api-client/src/main/java/com/xing/api/data/messages/ConversationMessage.java:17]: Needs licence header.

File: api-client/src/main/java/com/xing/api/data/messages/ConversationMessage.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        }
+
+        return (messageId.equals(m.messageId));
+    }
+
+    @Override
+    public int hashCode() {
+        return messageId != null ? messageId.hashCode() : 0;
+    }
+
```
Comment (serj-lotutovici) [inline api-client/src/main/java/com/xing/api/data/messages/ConversationMessage.java:61]: `toString()`?

File: api-client/src/main/java/com/xing/api/data/messages/MessageAttachment.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
+package com.xing.api.data.messages;
```
Comment (serj-lotutovici) [inline api-client/src/main/java/com/xing/api/data/messages/MessageAttachment.java:17]: Licence & `toString()`

File: api-client/src/main/java/com/xing/api/resources/MessagesResource.java [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+     * @param content Message text with max size of 16384 UTF-8 characters
+     * @param recipientIds Comma-separated list of recipients. There must be between one and 10 recipients.
+     * Sender cannot be included.
+     * @param subject Subject for conversation. Max. size is 255 UTF-8 characters
+     * @param userId Id of the {@linkplain com.xing.api.data.profile.XingUser user} starting the conversation
+     * @return A {@linkplain CallSpec callSpec object} ready to execute the request.
+     *
+     * @see <a href="https://dev.xing.com/docs/post/users/:user_id/conversations">'Create a conversation' resource page</a>
+     */
+    public CallSpec<Conversation, HttpError> createConversation(String content, String recipientIds, String subject,
```
Comment (serj-lotutovici) [inline api-client/src/main/java/com/xing/api/resources/MessagesResource.java]: Please change the signature to `createConversation(String userId, String subject, String content, String... recipientIds)`, Technically a user should not be able to send a message on behalf of another user, so I'm thinking that userId may be redundant here.

Comment (serj-lotutovici) [inline api-client/src/main/java/com/xing/api/resources/MessagesResource.java]: Ok, checked the docks, and it seams that a request with url `/v1/users/me/conversations` is not supported... just change the signature..

File: api-client/src/main/java/com/xing/api/resources/MessagesResource.java [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+     *
+     * @param messageContent The {@linkplain ConversationMessage message} to add
+     * @param conversationId Id of the {@link Conversation conversation} where the message is created in
+     * @param userId Id of the {@linkplain com.xing.api.data.profile.XingUser user} who sends the message
+     * @return A {@linkplain CallSpec callSpec object} ready to execute the request.
+     *
+     * @see <a href="https://dev.xing.com/docs/post/users/:user_id/conversations/:conversation_id/messages">'Create
+     * message in a conversation' resource page</a>
+     */
+    public CallSpec<ConversationMessage, HttpError> sendMessageToConversation(String messageContent,
```
Comment (serj-lotutovici) [inline api-client/src/main/java/com/xing/api/resources/MessagesResource.java]: Same here... should be `sendMessageToConversation(userId, conversationId, messageContent)`

**Comment by serj-lotutovici** [comment]
LGTM. Just a few comments.

**Pull request merged**